wallet-new/src/Cardano/Wallet/Kernel/DB/AcidState.hs

```diff
@@ -104,6 +104,18 @@ newPending accountId tx = runUpdate' . zoom dbHdWallets $
     zoom hdAccountCheckpoints $
       mapUpdateErrors NewPendingFailed $ Spec.newPending tx
 
+-- | Cancels the input transactions from all the 'Checkpoints' of each of
+-- the accounts cointained in all the wallets.
+--
+-- The reason why this function doesn't take a 'HdRootId' as an argument
+-- is because the submission layer doesn't have the notion of \"which HdWallet
+-- is this transaction associated with?\", but it merely dispatch and cancels
+-- transactions for all the wallets managed by this edge node.
+cancelPending :: InDb (Set Core.TxId) -> Update DB ()
+cancelPending tx =
+    runUpdateNoErrors . zoomAll (dbHdWallets . hdWalletsAccounts) $
+        hdAccountCheckpoints %~ Spec.cancelPending tx
```

Comment:
I was thinking about this when I couldn't sleep. There shouldn't be a need to update _all_ accounts, right? Each transaction is only submitted to a _single_ one.